### ISSUE TITLE
azure-cli homebrew release process update request

### ISSUE THREAD

**Issue by chenrui333** [Azure/azure-cli#16388]
👋  per discussions in [here](https://github.com/Homebrew/homebrew-core/pull/67930#discussion_r549572566), you folks need to update the homebrew release process.

Also, I wonder if it would be possible to publish that big source archive to pypi and use that instead. Thanks!

**Comment by fengzhou-msft** [comment]
Thanks for submitting the issue. We'll work on it during the next Sprint.

**Comment by carlocab** [comment]
FYI, it would be useful to update the URL to PyPI to enable the automation of resource updates using `brew update-python-resources`. The current azure-cli formula seems to have a large number of outdated resources:

https://github.com/carlocab/homebrew-core/commit/57dbe4a285e2af1c5a43e70dadacb20d5cc0cb53

One outdated resource caused this problem: https://github.com/Homebrew/brew/issues/10152#issuecomment-753334171

I fixed that particular problem here https://github.com/Homebrew/homebrew-core/pull/68126, but there's no telling what other problems the still outdated resources could cause.

**Comment by fengzhou-msft** [comment]
@carlocab we used `az.completion` file from our current source code tarball and it's not in the `azure-cli` pypi source package. Is there some easy ways to use the `az.completion` file from some URL in the formula if we switch to use the pypi resource for `azure-cli` packages?

**Comment by carlocab** [comment]
Hmm, one way might be to declare `az.completion` in a `resource` block, but I think that might break the automation in `update-python-resources`, as I believe it only works when all the resources come from PyPI.

@Rylan12, thoughts?

**Comment by Rylan12** [comment]
Yeah, adding as a resource wouldn't cooperate well with `update-python-resources` in its current form. There might be another way we can get around that, but I can't think of it off the top of my head.

One thing that some formulae do is have a separate completion formula. So a user could `brew install azure-cli` but would need to `brew install azure-cli-completion` if they wanted completion. Probably not ideal as a user would have to install two formulae. Although we could probably put a message in the caveats instructing users to install the completions formula if desired.

**Comment by fengzhou-msft** [comment]
@carlocab @Rylan12 thanks for the suggestions. I'm also thinking about moving the `az.completion` file inside the `azure-cli` directory, and package it the `azure-cli` tarball, then I assume it could be accessed in the install step?

**Comment by Rylan12** [comment]
Assuming the file is in the source tarball that's downloaded from PyPI (the "Source" file type at https://pypi.org/project/azure-cli/#files), it should be accessible in the install method.

As for how to ensure that it makes it into the tarball, I'm really not sure. It looks like the contents of the tarball nearly matches the [`src/azure-cli/` directory](https://github.com/Azure/azure-cli/tree/dev/src/azure-cli), so that maybe that's where it goes (maybe that's what you're suggesting). There already is a `az.completion.sh` file there, so that seems like an appropriate place I think.

Not sure if that's helpful. You know a million times more about `azure-cli` than I do so it's hard for me to weigh in. The only thing I can say for certain is that if it makes it in the PyPI tarball and that's specified as the formula's `url`, we will be able to access it in the install method.
